### ISSUE TITLE
is there a way to start and terminate an event listener ? 

### ISSUE THREAD

**Issue by nemr** [node-enocean/node-enocean#36]
Hi , 

Thanks for this amazing library.

I'm trying to add a listener  by using the enocean.on("data", function (data) {});

is there way to remove that listener as I only want it to listen for a specific period of time ?

**Comment by Holger-Will** [comment]
As the enocean object inherits from EventEmitter, it should be straight forward...

    var listener= function(data){}
    enocean.on("data",listener)
    enocean.removeListener(listener)

i haven't tested yet, and can not test right now, but it should theoretically work like that.
please see the documentation of EventEmitter for all inherited Methods: https://nodejs.org/api/events.html

**Comment by Holger-Will** [comment]
now found the time to test it... this way it works as intented:

    var listener= function(data){}
    enocean.on("data",listener)
    enocean.removeListener("data",listener)

you can also remove anonymus listeners:

    enocean.on("data",function(data){})
    enocean.removeListener("data",enocean.listeners("data")[0])

**Comment by nemr-maher** [comment]
Thanks , that exactly what I was looking for 👍 :)

**Comment by nemr** [comment]
Hi @Holger-Will .

Thanks for your support, I just have a question which is related to this topic.

I'm trying to send a command using enocean.send but I would like to wait for a response, so I would have to use enocean.on("data",function(data){}).

So what I have atm is something like this :
```
   function sendCommand(){
                    enocean.send ("EnOcean Command ")
                        enocean.on("data", function (data) {
                            if (data.senderId == "xxxxxxx") {
                                console.log(data);

                                enocean.removeListener("data", enocean.listeners("data")[0])
                                return;
                            }
                        })
}
```
I also have enocean.on("data",function(data){}) used in my script in general (to store sensor values).

is there a better way of implementing this using your npm? where I basically want to be able to listen for enocean events all the time but would still like to use something such as EnOcean.on("data",function(data){})  within a function.

So Basically being able to access data.senderId anywhere in my code without the need to add a listener each time.

Kind Regards
